lib/store.ml

```diff
@@ -43,10 +51,23 @@ struct
   (** Store are represented as arrays. For small values, which is typicial,
       is slightly quicker than other implementations. *)
 
-type 'a field = int
+
+  type 'a merge_arg =
+    | Left of 'a
+    | Right of 'a
```

Comment:
It is kind of annoying that we need the case with a single `'a` at all. Since it precludes optimisations. But well.

On the other hand I think we really do _not_ want an operation that distinguishes `Left` and `Right` (I don't think there is a guarantee for the ordering of equations it would be another nice way to have instabilities in the unification algorithm). So I'd suggest:
```ocaml
type 'a merge_arg =
  | One of 'a
  | Both of 'a*'a
```